homeassistant/components/lawn_mower/__init__.py

```diff
@@ -0,0 +1,162 @@
+"""The lawn mower integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+from enum import IntFlag
+from functools import partial
+import logging
+from typing import Any, final
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.config_validation import (  # noqa: F401
+    PLATFORM_SCHEMA,
+    PLATFORM_SCHEMA_BASE,
+)
+from homeassistant.helpers.entity import Entity, EntityDescription
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_START_MOWING = "start_mowing"
+SERVICE_PAUSE = "pause"
+SERVICE_ENABLE_SCHEDULE = "enable_schedule"
+SERVICE_DISABLE_SCHEDULE = "disable_schedule"
+SERVICE_DOCK = "dock"
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the lawn_mower component."""
+    component = hass.data[DOMAIN] = EntityComponent[LawnMowerEntity](
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL
+    )
+    await component.async_setup(config)
+
+    component.async_register_entity_service(
+        SERVICE_START_MOWING,
+        {},
+        "async_service_start_mowing",
+        [LawnMowerEntityFeature.START_MOWING],
+    )
+    component.async_register_entity_service(
+        SERVICE_PAUSE, {}, "async_service_pause", [LawnMowerEntityFeature.PAUSE]
+    )
+    component.async_register_entity_service(
+        SERVICE_ENABLE_SCHEDULE,
+        {},
+        "async_service_enable_schedule",
+        [LawnMowerEntityFeature.ENABLE_SCHEDULE],
+    )
+    component.async_register_entity_service(
+        SERVICE_DISABLE_SCHEDULE,
+        {},
+        "async_service_disable_schedule",
+        [LawnMowerEntityFeature.DISABLE_SCHEDULE],
+    )
+    component.async_register_entity_service(
+        SERVICE_DOCK, {}, "async_service_dock", [LawnMowerEntityFeature.DOCK]
+    )
+
+    return True
+
+
+class Activity(IntFlag):
+    """Activity state of the lawn mower entity."""
+
+    ERROR = 1
+    PAUSED = 2
+    MOWING = 4
+    DOCKED_SCHEDULE_DISABLED = 8
+    DOCKED_SCHEDULE_ENABLED = 16
+
+
+class LawnMowerEntityFeature(IntFlag):
+    """Supported features of the lawn mower entity."""
+
+    START_MOWING = 1
+    PAUSE = 2
+    DOCK = 4
+    ENABLE_SCHEDULE = 8
+    DISABLE_SCHEDULE = 16
```

Comment:
Isn't Schedule one feature instead of two separately for enable/disable?

Reply:
If it's two different services and we don't require both services, it's safer to have two features.